### DISCUSSION TITLE
`CONFIGURE_OPTS="--disable-install-rdoc" rbenv install 3.2.0` not working

### DISCUSSION THREAD

**Discussion by zdavatz** [rbenv/ruby-build#2122] [Build failures] [answered]
1. I am trying to do: `CONFIGURE_OPTS="--disable-install-rdoc" rbenv install 3.2.0`
2. But this results in:
```
To follow progress, use 'tail -f /tmp/ruby-build.20230103212239.433590.log' or pass --verbose
Downloading ruby-3.2.0.tar.gz...
-> https://cache.ruby-lang.org/pub/ruby/3.2/ruby-3.2.0.tar.gz
Installing ruby-3.2.0...

BUILD FAILED (Ubuntu 22.04 using ruby-build 20221225-9-g697bcff)

Inspect or clean up the working tree at /tmp/ruby-build.20230103212239.433590.YVjXF6
Results logged to /tmp/ruby-build.20230103212239.433590.log

Last 10 log lines:
	from ./tool/rbinstall.rb:721:in `no_write'
	from ./tool/rbinstall.rb:881:in `install'
	from ./tool/rbinstall.rb:989:in `block in install_default_gem'
	from ./tool/rbinstall.rb:976:in `each'
	from ./tool/rbinstall.rb:976:in `install_default_gem'
	from ./tool/rbinstall.rb:913:in `block in <main>'
	from ./tool/rbinstall.rb:1077:in `block in <main>'
	from ./tool/rbinstall.rb:1074:in `each'
	from ./tool/rbinstall.rb:1074:in `<main>'
make: *** [uncommon.mk:416: do-install-nodoc] Error 1
````

**Comment by hsbt** [comment, marked as answer]
You should use `--disable-install-doc` instead of `--disable-install-rdoc`.

**Comment by jarthod** [comment]
For the record and for future people finding this post after having trouble with this option, I recently encountered a similar error while compiling Ruby 3.1.2 on Ubuntu 24.04.1 LTS (recently updated). The log file contained this error message:
```
cc1: error: unknown pass install-doc specified in '-fdisable'
cc1: error: unknown pass install-doc specified in '-fdisable'
```

I tried with `--disable-install-doc` and `--disable-install-rdoc` but the result was the same. What I guessed is that newer versions of `cc` or `make` might be more pendentic and refusing this option where it's not expected (I never had the problem with this syntax using Ubuntu 22.04 and earlier). So I tried passing the options to `RUBY_CONFIGURE_OPTS` instead of `CONFIGURE_OPTS`, thinking this might be more specific and not "leaking" as much.

Seems to be working fine for me now with:
```console
$ RUBY_CONFIGURE_OPTS="--disable-install-doc" rbenv install 3.1.2
```
If you already have option like me, you can separate them by spaces as long as you put quotes outside the variable:
```console
$ RUBY_CONFIGURE_OPTS="--with-jemalloc --disable-install-doc" rbenv install 3.1.2
```
